packages/usa-input-mask/src/index.js

```diff
@@ -1,124 +1,1095 @@
 const selectOrMatches = require("../../uswds-core/src/js/utils/select-or-matches");
 const behavior = require("../../uswds-core/src/js/utils/behavior");
+const debounce = require("../../uswds-core/src/js/utils/debounce");
 const { prefix: PREFIX } = require("../../uswds-core/src/js/config");
 
-const MASKED_CLASS = `${PREFIX}-masked`;
-const MASKED = `.${MASKED_CLASS}`;
-const MASK = `${PREFIX}-input-mask`;
-const MASK_CONTENT = `${MASK}--content`;
+const MASKED_INPUT_CLASS = `${PREFIX}-input-mask`;
+const MASKED_INPUT = `.${MASKED_INPUT_CLASS}`;
+const MASKED_INPUT_SHELL_CLASS = `${MASKED_INPUT_CLASS}__shell`;
+const MASKED_INPUT_CONTENT_CLASS = `${MASKED_INPUT_CLASS}__content`;
 const PLACEHOLDER = "placeholder";
-const CONTEXT = "form";
-
-// User defined Values
-const maskedNumber = "_#dDmMyY9";
-const maskedLetter = "A";
-
-// replaces each masked input with a shell containing the input and it's mask.
-const createMaskedInputShell = (input) => {
-  const placeholder = input.getAttribute(`${PLACEHOLDER}`);
-  if (placeholder) {
-    input.setAttribute("maxlength", placeholder.length);
-    input.setAttribute("data-placeholder", placeholder);
-    input.removeAttribute(`${PLACEHOLDER}`);
+
+const MESSAGE_INVALID_CLASS = `${PREFIX}-error-message`;
```

Comment:
I see that there are four potential options for error styling here. Error styling can be added to `usa-form-group`, `usa-label`, `usa-input` and/or `usa-input-mask__status`. I went with just styling the status message because it is something that flagged with each invalid keystroke, instead of after the user either completes the input or form. For that reason, I wanted to make the styles as minimally disruptive as possible. Adding error styles to only the status message allows movement on only the vertical axis and not the horizontal. 

Here is what the component would look like with few different levels of error styling. Let me know if you have a preference.

![image](https://user-images.githubusercontent.com/93996430/235244239-7f1ede69-02c2-4062-ba93-cfc592337811.png)